src/Microsoft.AspNet.Mvc.Core/ContentResult.cs

```diff
@@ -64,7 +59,7 @@ public override Task ExecuteResultAsync(ActionContext context)
                 var bufferingFeature = response.HttpContext.Features.Get<IHttpBufferingFeature>();
                 bufferingFeature?.DisableResponseBuffering();
 
-                return response.WriteAsync(Content, contentTypeHeader?.Encoding ?? DefaultContentType.Encoding);
+                return response.WriteAsync(Content, contentTypeHeader.Encoding);
```

Comment:
This doesn't do what we want. If the user set an explicit content type that doesn't include `charset`, we're aren't trying to 'fix' it for them.